bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/ObservationWeatherHandler.java

```diff
@@ -0,0 +1,193 @@
+ *
+ * @author Sami Salonen - Initial contribution
+ */
+@NonNullByDefault
+public class ObservationWeatherHandler extends AbstractWeatherHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ObservationWeatherHandler.class);
+    private static final long OBSERVATION_LOOK_BACK_SECONDS = TimeUnit.MINUTES.toSeconds(30);
+    private static final int STEP_MINUTES = 10;
+    private static final int POLL_INTERVAL_SECONDS = 600;
+
+    public static final Unit<Length> MILLIMETRE = MetricPrefix.MILLI(METRE);
+    public static final Unit<Length> CENTIMETRE = MetricPrefix.CENTI(METRE);
+
+    private static final Map<String, Map.Entry<String, @Nullable Unit<?>>> CHANNEL_TO_OBSERVATION_FIELD_NAME_AND_UNIT = new HashMap<>(
+            11);
+
+    private static void addMapping(String channelId, String requestField, @Nullable Unit<?> unit) {
+        CHANNEL_TO_OBSERVATION_FIELD_NAME_AND_UNIT.put(channelId,
+                new AbstractMap.SimpleImmutableEntry<>(requestField, unit));
+    }
+
+    static {
+        addMapping(CHANNEL_TEMPERATURE, PARAM_TEMPERATURE, CELSIUS);
+        addMapping(CHANNEL_HUMIDITY, PARAM_HUMIDITY, PERCENT);
+        addMapping(CHANNEL_WIND_DIRECTION, PARAM_WIND_DIRECTION, DEGREE_ANGLE);
+        addMapping(CHANNEL_WIND_SPEED, PARAM_WIND_SPEED, METRE_PER_SECOND);
+        addMapping(CHANNEL_GUST, PARAM_WIND_GUST, METRE_PER_SECOND);
+        addMapping(CHANNEL_PRESSURE, PARAM_PRESSURE, MILLIBAR);
+        addMapping(CHANNEL_PRECIPITATION_AMOUNT, PARAM_PRECIPITATION_AMOUNT, MILLIMETRE);
+        addMapping(CHANNEL_SNOW_DEPTH, PARAM_SNOW_DEPTH, CENTIMETRE);
+        addMapping(CHANNEL_VISIBILITY, PARAM_VISIBILITY, METRE);
+        addMapping(CHANNEL_CLOUDS, PARAM_CLOUDS, null);
+        addMapping(CHANNEL_OBSERVATION_PRESENT_WEATHER, PARAM_PRESENT_WEATHER, null);
+    }
+
+    private @NonNullByDefault({}) String fmisid;
+
+    public ObservationWeatherHandler(Thing thing) {
+        super(thing);
+        pollIntervalSeconds = POLL_INTERVAL_SECONDS;
+    }
+
+    @Override
+    public void initialize() {
+        fmisid = getConfig().get(BindingConstants.FMISID).toString();
+        super.initialize();
+    }
+
+    @Override
+    protected void update() {
+        for (int retry = 0; retry < RETRIES; retry++) {
+            try {
+                long now = Instant.now().getEpochSecond();
+                response = client.query(new ObservationRequest(new FMISID(fmisid),
+                        floorToEvenMinutes(now - OBSERVATION_LOOK_BACK_SECONDS, STEP_MINUTES),
+                        ceilToEvenMinutes(now, STEP_MINUTES), STEP_MINUTES), TIMEOUT_MILLIS);
+            } catch (FMIResponseException e) {
+                response = null;
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        String.format("%s: %s", e.getClass().getName(), e.getMessage()));
+                try {
+                    Thread.sleep(RETRY_DELAY_MILLIS);
```

Comment:
I'm not sure how that would work out really since this method is called with `scheduleWithFixedDelay`.

Assuming exceptions are rare, I find this current approach acceptable.

Reply:
Implemented in e455d438856140cd7f0d3435f5165e59af3ddec3 , appreciate second pair of eyes